### DISCUSSION TITLE
Component Menu, trigger event handler on MenuItem when MenuItem is activated

### DISCUSSION THREAD

**Discussion by dohaicuong** [chakra-ui/ark#2477] [Q&A] [answered]
With the current behavior, when user press `enter` or `space` on highlight menu item (keyboard navigation), the item will be activated and the menu is close hence `onSelect` is called on Menu.Root

Is there any way for me to trigger event on Menu.Item itself?

At the moment, passing `onClick` to Menu.Item works if user actually click on it,
but it's not trigger with keyboard `enter` or `space`.
I also tried `onChange`, `onSelect` and `onKeyDown` on Menu.Item but nothing got triggered.

**Comment by segunadebayo** [comment, marked as answer]
I suppose you can use the `onSelect`callback to trigger and event on the item yourself.

```js
function onSelect(details){
  const item = document.querySelector(`[data-value=${details.value}]`)
  item.click() // or item.dispatchEvent(new CustomEvent('itemselect', { details }))
}
```

**Reply by dohaicuong** [reply]
that's what I'm using at the moment, just wonder if there will be a fix to make this more straight forward to use
